sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -189,4 +191,6 @@ class CSVOptions(
     settings.setUnescapedQuoteHandling(UnescapedQuoteHandling.STOP_AT_DELIMITER)
     settings
   }
+
+  val textOptions = ListMap(parameters.toList: _*)
```

Comment:
Follow `JSONOptions` to make it `@transient` and use `parameters` instead of creating a new map?